tests/installation/accept_license.pm

```diff
@@ -34,8 +34,11 @@ sub run {
             die 'It seems that license agreement is missing, please check!';
         }
     }
-    $self->verify_license_has_to_be_accepted;
+  VERIFY: $self->verify_license_has_to_be_accepted;
     send_key $cmd{next};
+    if (check_screen('license-not-accepted', 1)) {    # workaround for bsc#1059317, sometimes need to click accept license more times
+        goto VERIFY;
```

Comment:
using goto is bad programming practice
what about following : 

```
 $count = 10;
do {
 $self->verify_license_has_to_be_accepted;
 $count--;
}while(check_screen('license-not-accepted', 1) && count >= 1);
```
 Of course 10 is just random number you need to think how many times you want to repeat this before fail . BTW in your current version you also in danger of indefinite loop

Reply:
ah and of course you need apply valid comment from @okurz  about soft failure in case of workaround for existing bug

Reply:
the indefinite loop is a valid concern, the usage of goto is not. goto has to be used careful, but using it is definitely an option and not per se bad.